### PR TITLE
Extend cacl function test

### LINKED ISSUE #3597: Need new test: System and Management - Access-control lists (ACL) on NTP protocol
This is a test gap:

Need to 1. generate NTP acl table in minigraph. 2. generate acl.json

### PR DESCRIPTION
**Pull request by AndoniSanguesa**
Signed-Off By: t-asanguesa@microsoft.com

### Description of PR

Added NTP capability to the existing Control Plane ACL functional test. Previously only SSH and NTP were supported. 

The docker container [python dependencies](https://github.com/Azure/sonic-buildimage/blob/master/dockers/docker-sonic-mgmt/Dockerfile.j2) defined by the `sonic-buildimage` repo does not include `ntplib` which is used by this addition. The parts of the test that use `ntplib` are disabled if the `ntplib` module is not present.

Summary:
Fixes #3597 

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911

#### What is the motivation for this PR?

There was no existing support for NTP in existing cacl function test.  

#### How did you do it?
Configured NTP ACL table in the minigraph xml files. Modified existing cacl config file to configure the new NTP ACL table. Sent NTP requests with and without the CACL configured.

Also made some changes to timeouts and added optional timeout argument to the snmp_facts ansible module. This change greatly improved test stability locally.

#### How did you verify/test it?
Ran tests multiple times with no errors. Manually checked that the CACL configuration was stopping communication on specified protocols via cli.

### Documentation 

Original test had no testplan. I added a very simple one while I made these modifications.

### CODE REVIEW

**Comment by lguohan** [comment]
can you add this to the kvmtest.sh? the test is control plane test.

**Review by yxieca — APPROVED** [review]

**Comment by yxieca** [comment]
> can you add this to the kvmtest.sh? the test is control plane test.

test_cacl_function.py is already part of the kvm test.

File: tests/cacl/test_cacl_function.py [hunk 1]
```diff
@@ -13,16 +21,29 @@
...
+
     duthost = duthosts[rand_one_dut_hostname]
     dut_mgmt_ip = duthost.mgmt_ip
 
+    if not NTPLIB_INSTALLED:
+        logging.warning("Will not check NTP connection. ntplib is not installed.")
+
+    # Start an NTP client
+    if NTPLIB_INSTALLED:
+        ntp_client = ntplib.NTPClient()
```
Comment (yxieca) [inline tests/cacl/test_cacl_function.py:34]: Use if else structure for line 29 - 34?

File: tests/cacl/test_cacl_function.py [hunk 2]
```diff
@@ -13,16 +21,29 @@
...
+
     # Ensure we can gather basic SNMP facts from the device
-    res = get_snmp_facts(localhost, host=dut_mgmt_ip, version='v2c', community=creds['snmp_rocommunity'])
+    res = localhost.snmp_facts(host=dut_mgmt_ip, version='v2c', community=creds['snmp_rocommunity'], timeout=20)
```
[the remark this answers is not in the document]
Comment (yxieca) [reply tests/cacl/test_cacl_function.py:37]: Line 113 in this file also have example of using the helper.

**Comment by AndoniSanguesa** [comment]
Thanks for the suggestions @yxieca. Just push those commits

**Pull request merged**